src/Dev/Deprecation.php

```diff
@@ -163,8 +183,59 @@ protected static function get_called_method_from_trace($backtrace, $level = 1)
         if ($level == 4 && ($backtrace[$newLevel]['class'] ?? '') === InjectionCreator::class) {
             $newLevel = $newLevel + 4;
         }
+        // handle noticeWithNoReplacment()
+        foreach ($backtrace as $trace) {
+            if (($trace['class'] ?? '') === Deprecation::class
+                && ($trace['function'] ?? '') === 'noticeWithNoReplacment'
+            ) {
+                $newLevel = $newLevel + 1;
+                break;
+            }
+        }
         $called = $backtrace[$newLevel] ?? [];
-        return ($called['class'] ?? '') . ($called['type'] ?? '') . ($called['function'] ?? '');
+        return $called;
+    }
+
+    private static function isCalledFromSupportedCode(array $backtrace): bool
+    {
+        $called = Deprecation::get_called_from_trace($backtrace, 1);
+        $file = $called['file'] ?? '';
+        if (!$file) {
+            return false;
+        }
+        return Deprecation::fileIsInSupportedModule($file);
+    }
+
+    /**
+     * Check whether a file (path to file) is in a supported module
+     */
+    public static function fileIsInSupportedModule(string $file): bool
+    {
+        // Cache the supported modules list
+        if (count(Deprecation::$supportedModuleDirectories) === 0) {
+            // Manually load the supported modules list rather than use MetaData::getAllRepositoryMetaData()
+            // because we do not want to make a network request which could slow down a website
+            // While there is a small risk of the list being out of date, there is minimal downside to this
+            $path = Path::join(BASE_PATH, 'vendor/silverstripe/supported-modules/repositories.json');
```

Comment:
Can we just update `MetaData::getAllRepositoryMetaData()` to include a boolean parameter so we can tell that API whether to do a network request or not? I'd much rather keep to using the PHP API in that module rather than bypassing it.

Reply:
Have added to supported-modules PR

Reply:
Can you please update this code to stop using the filepath directly?

Reply:
Forgot to push, updated